### PR TITLE
fix: Set AppWindow.Title to DisplayName

### PR DESCRIPTION
**Pull request by lukeblevins**
Fixes: #208 
Replaces #209 

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
On Windows targets, `AppWindow` currently lacks a default value for its `Title` property. 

In our task for generating window icon, we form the code to retrieve an `AppWindow` from the XAML `Window`, using it to conveniently set the generated `ico` file. 

## What is the new behavior?

Upon retrieval of `AppWindow`, set AppWindow.Title to be the app's `DisplayName`. This value is obtained from the XAML `Window` instance which has a pre-defined `Title` property. (defaulting to `DisplayName`)

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable): N/A

### CODE REVIEW

File: src/Resizetizer/src/WindowIconGeneratorTask.cs [hunk 1]
```diff
@@ -68,19 +79,41 @@ public static void SetWindowIcon(this global::Microsoft.UI.Xaml.Window window)
...
-			Directory.CreateDirectory(IntermediateOutputDirectory);
-		}
-
 		var item = new TaskItem(Path.Combine(IntermediateOutputDirectory, FileName));
 		File.WriteAllText(item.ItemSpec, code);
 		GeneratedClass = new [] { item };
 		return true;
 	}
+
+	string GetDisplayName()
```
Comment (pictos) [inline src/Resizetizer/src/WindowIconGeneratorTask.cs:94]: maybe the info that you want will can find it [here](https://github.com/unoplatform/uno.resizetizer/blob/main/src/Resizetizer/src/GeneratePackageAppxManifest.cs)

Comment (lukeblevins) [reply src/Resizetizer/src/WindowIconGeneratorTask.cs:94]: @pictos Yes, I modeled this fix after the implementation in that file

File: src/Resizetizer/src/WindowIconGeneratorTask.cs [hunk 2]
```diff
@@ -31,6 +31,17 @@ public override bool Execute()
...
 
+		var windowTitle = string.Empty;
+
+        if (Directory.Exists(IntermediateOutputDirectory))
+		{
+			windowTitle = GetDisplayName();
+        }
+		else
+		{
+            Directory.CreateDirectory(IntermediateOutputDirectory);
```
Comment (nickrandolph) [inline src/Resizetizer/src/WindowIconGeneratorTask.cs:42]: Why is it necessary to create the directory if it doesn't exist?
Can you check spacing as it looks inconsistent here

File: src/Resizetizer/src/WindowIconGeneratorTask.cs (hunk 1 above)
Comment (nickrandolph) [inline src/Resizetizer/src/WindowIconGeneratorTask.cs:94]: Methods require accessor
```suggestion
	private string GetDisplayName()
```

File: src/Resizetizer/src/WindowIconGeneratorTask.cs [hunk 3]
```diff
@@ -68,19 +79,41 @@ public static void SetWindowIcon(this global::Microsoft.UI.Xaml.Window window)
...
-
 		var item = new TaskItem(Path.Combine(IntermediateOutputDirectory, FileName));
 		File.WriteAllText(item.ItemSpec, code);
 		GeneratedClass = new [] { item };
 		return true;
 	}
+
+	string GetDisplayName()
+	{
+		string manifestOutputPath = Path.Combine(IntermediateOutputDirectory, "m", "Package.appxmanifest");
```
Comment (nickrandolph) [inline src/Resizetizer/src/WindowIconGeneratorTask.cs:96]: Can we rationalise the code that loads the manifest, so we're not parsing it multiple times?

File: src/.nuspec/Uno.Resizetizer.targets [hunk 4]
```diff
@@ -513,6 +513,7 @@
 			Outputs="@(_WindowIconExtension)">
 		<WindowIconGeneratorTask_v0
 				IntermediateOutputDirectory="$(_UnoResizetizerIntermediateOutputRoot)"
+				WindowTitle="$(ApplicationTitle)"
```
Comment (nickrandolph) [inline src/.nuspec/Uno.Resizetizer.targets:516]: @pictos is this the correct way to retrieve the application title? And will this result in the same value used for other platforms where the title should come from packageappx.manifest?

Comment (pictos) [reply src/.nuspec/Uno.Resizetizer.targets:516]: @nickrandolph isn't the correct way. I talked with Luke on Teams showing him how he can retrieve the `ApplicationTitle` property. Then is up to him if he wants to expose it on MSBuild tasks or reference on code, using some static property.

File: src/Resizetizer/src/GeneratePackageAppxManifest.cs [hunk 5]
```diff
@@ -45,6 +45,9 @@ public class GeneratePackageAppxManifest_v0 : Task
 		[Output]
 		public ITaskItem GeneratedAppxManifest { get; set; } = null!;
 
+		[Output]
+		public string DisplayName { get; private set; } = null!;
```
Comment (nickrandolph) [inline src/Resizetizer/src/GeneratePackageAppxManifest.cs:49]: Is there a reason we don't just let this property be nullable? ie string?

Comment (nickrandolph) [reply src/Resizetizer/src/GeneratePackageAppxManifest.cs:49]: And why do we need both ApplicationTitle and DisplayName? Won't they be the same?

File: src/Resizetizer/src/GeneratePackageAppxManifest.cs [hunk 6]
```diff
@@ -143,19 +146,22 @@ void UpdateManifest(XDocument appx)
...
-						properties.SetElementValue(xname, ApplicationTitle);
-					}
-				}
+						properties.SetElementValue(xDisplayName, ApplicationTitle);
+                    }
+                }
 
-				// <Logo>
-				if (appIconInfo != null)
+				DisplayName = xDisplayNameElem.Value;
```
Comment (nickrandolph) [inline src/Resizetizer/src/GeneratePackageAppxManifest.cs:161]: Shouldn't this be ApplicationTitle?

Comment (pictos) [reply src/Resizetizer/src/GeneratePackageAppxManifest.cs]: No, looking further the `ApplicationTitle` is an existing property that can be set on projects... Specifically here, is used on [Mobile targets](https://github.com/unoplatform/uno.resizetizer/blob/cf892d2932cc30e620879d88181412f831ddcae8/samples/NewTemplate/Resizetizer.Extensions.Sample.Mobile/Resizetizer.Extensions.Sample.Mobile.csproj#L7) and we can't assume that will not be used on other targets (skia, wasm, etc)

**Review by nickrandolph — APPROVED** [review]

**Pull request merged**